src/clj/ataru/hakija/hakija_form_service.clj

```diff
@@ -53,33 +53,57 @@
           :public-config
           (get :attachment-modify-grace-period-days 14))))
 
+(defn- get-hakukohteiden-hakuajat-for-oids [hakukohde-oids hakukohteet]
+       (let [belongs-to (set hakukohde-oids)
+             filters [#(contains? belongs-to (:oid %))
+                      #(some? (:hakuaika-dates %))]]
+            (if-let [hakukokohde-hakuajat (filter (fn [x] (every? #(% x) filters)) hakukohteet)]
+                    (map :hakuaika-dates hakukokohde-hakuajat))))
+
+(defn- select-first-ongoing-hakuaika-or-hakuaika-with-last-ending [hakuajat]
+  (if-let [ongoing-hakuaika (first (filter :on hakuajat))]
+    ongoing-hakuaika
+    (first (sort-by :end > hakuajat))))
+
+(defn- select-hakuaika-for-field [field hakukohteet haun-hakuaika]
+  (let [hakukohteet-with-hakuajat (filter #(some? (:hakuaika-dates %)) hakukohteet)
+        all-hakuajat (concat [haun-hakuaika] (map :hakuaika-dates hakukohteet-with-hakuajat))
+        belongs-to-hakukohteet (if-let [b (seq (:belongs-to-hakukohteet field))] b)
+        belonging-hakuajat (some-> belongs-to-hakukohteet (get-hakukohteiden-hakuajat-for-oids hakukohteet-with-hakuajat))]
```

Comment:
Maybe use when instead of if-let and some-> on lines 71 ad 72?
71: (when (not-empty (:belongs...))
72: (when belongs to hakukohteet ...)

Reply:
https://clojuredocs.org/clojure.core/empty_q
"Please use the idiom (seq x) rather than (not (empty? x))" is the preferred way to do (not (empty? ...))

Reply:
But i agree that not-empty is more readable